analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -520,3 +521,26 @@ class Pub(
         run(workingDir, "get")
     }
 }
+
+/**
+ * Extract information about package authors from the given [pubspec].
+ */
+private fun fetchAuthors(pubspec: JsonNode): SortedSet<String> {
+    val authors = sortedSetOf<String>()
+    extractAuthorName(pubspec["author"].textValueOrEmpty())?.let { authors += it }
+
+    pubspec["authors"]?.forEach { authorNode ->
+        extractAuthorName(authorNode.textValueOrEmpty())?.let { authors += it }
+    }
```

Comment:
Could probably be rewritten to something like

    (pubspec["authors"].toList() + pubspec["author"]).mapNotNullTo(authors) {
        extractAuthorName(it.textValue())
    }

(if `extractAuthorName()` was also changed to accept a nullable string).

Reply:
I changed the function to a single expression, but it was not that easy. Interestingly, the expression `(pubspec["authors"].toList() + pubspec["author"])` yields an empty list if the _author_ field is defined. This is probably the case because `JsonNode` implements `Iterable`, and therefore a wrong overloaded variant of the `+` operator is used. I also had to add more null-checking logic.